### DISCUSSION TITLE
请问Xray(VLESS+XHTTP)+Nginx\Caddy过CDN不通，直连能通

### DISCUSSION THREAD

**Discussion by godspirit00** [lxhao61/integrated-examples#231] [Q&A] [answered]
请问大佬，参考Xray(VLESS+XHTTP)+Nginx\Caddy搭了一个同时存在有VMESS+WS+Caddy和Xray(VLESS+XHTTP)+Caddy的，通过CF CDN的话VMESS+WS+Caddy能通，Xray(VLESS+XHTTP)+Caddy不能，直连的话能通，请问可能是什么原因？

**Comment by lxhao61** [comment, marked as answer]
你应该没有看官方 XHTTP 资料及本仓库客户端示例提示！
具体情况是你 CF CDN 没有开启 gRPC 支持、且客户端 ``mode`` 配置选择的是 auto（默认）或 stream-up，这样经过 CDN 流量中转 XHTTP 应用肯定无法连通。你 CF CDN 可开启 gRPC 支持，这样客户端 ``mode`` 配置可随便选择；或者客户端 ``mode`` 配置仅选择 packet-up。

**Reply by godspirit00** [reply]
谢谢大佬！

**Comment by ArcCal** [comment]
好复杂

**Reply by maryjeck** [reply]
没套cf,怎么都连不上哦，xhttp不管选哪个模式都连不上

![image](https://github.com/user-attachments/assets/fcf77bce-83a9-4093-8329-e8008a823f79)
